pkg/machine/applehv/machine.go

```diff
@@ -679,20 +669,15 @@ func (m *MacMachine) Start(name string, opts machine.StartOptions) error {
 	logrus.Debug("waiting for ready notification")
 	var conn net.Conn
 	readyChan := make(chan error)
-	go func() {
-		conn, err = readyListen.Accept()
-		if err != nil {
-			logrus.Error(err)
-		}
-		_, err = bufio.NewReader(conn).ReadString('\n')
-		readyChan <- err
-	}()
+	connChan := make(chan net.Conn)
+	go machine.ListenAndWaitOnSocket(readyChan, connChan, readyListen)
 
 	if err := cmd.Start(); err != nil {
 		return err
 	}
 
 	err = <-readyChan
+	conn = <-connChan
```

Comment:
same as the other comment this can be nil so the close below is not safe.

Reply:
Done as well.